underlay/src/main/java/bio/terra/tanagra/underlay/entitygroup/CriteriaOccurrence.java

```diff
@@ -85,25 +87,35 @@ public static CriteriaOccurrence fromSerialized(
 
     // Relationships.
     Map<String, Relationship> relationships =
-        Map.of(
-            OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME,
-            new Relationship(
+        new HashMap<>(
+            Map.of(
                 OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME,
-                occurrenceEntity,
-                criteriaEntity,
-                buildRelationshipFieldList(criteriaEntity)),
-            OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME,
-            new Relationship(
+                new Relationship(
+                    OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME,
+                    occurrenceEntity,
+                    criteriaEntity,
+                    buildRelationshipFieldList(criteriaEntity)),
                 OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME,
-                occurrenceEntity,
-                primaryEntity,
-                Collections.emptyList()),
-            CRITERIA_TO_PRIMARY_RELATIONSHIP_NAME,
-            new Relationship(
+                new Relationship(
+                    OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME,
+                    occurrenceEntity,
+                    primaryEntity,
+                    Collections.emptyList()),
                 CRITERIA_TO_PRIMARY_RELATIONSHIP_NAME,
-                criteriaEntity,
-                primaryEntity,
-                buildRelationshipFieldList(criteriaEntity)));
+                new Relationship(
+                    CRITERIA_TO_PRIMARY_RELATIONSHIP_NAME,
+                    criteriaEntity,
+                    primaryEntity,
+                    buildRelationshipFieldList(criteriaEntity))));
+    if (visitOccurrenceEntity != null) {
```

Comment:
I didn't realize this was going to make visit occurrences part of the code. That may not generalize very well to other data sets that aren't in the same format. Would it be possible to list related entities in the underlay?

Reply:
Good point, thanks! I changed

```
Entity visitOccurrenceEntity
```

to 

```
List<Entity> occurrenceRelatedEntities
```

It's a list because there may be more than one entity related to occurrence entity. For example in OMOP, condition_occurrence is related to visit_occurrence and provider.

I reindexed and manually retested condition_person_occurrence.